src/framework/mlt_property.c

```diff
@@ -1279,7 +1281,50 @@ int mlt_property_interpolate( mlt_property self, mlt_property p[],
 	double progress, double fps, locale_t locale, mlt_keyframe_type interp )
 {
 	int error = 0;
-	if ( interp != mlt_keyframe_discrete &&
+	if ( interp != mlt_keyframe_discrete && ( self->types & mlt_prop_color ) )
+	{
+		mlt_color value = { 0xff, 0xff, 0xff, 0xff };
+		if ( interp == mlt_keyframe_linear )
+		{
+			mlt_color colors[2];
+			mlt_color zero = { 0xff, 0xff, 0xff, 0xff };
+			colors[0] = p[1]? mlt_property_get_color( p[1], fps, locale ) : zero;
+			if ( p[2] )
+			{
+				colors[1] = mlt_property_get_color( p[2], fps, locale );
```

Comment:
This should work OK. But might surprise some people since it will not naturally go "around the rainbow". For example, to get from green to yellow, it might go through brown. I wonder if we should convert both colors to HSL, interpolate the HSL values, and then convert back to RGB. This is not a blocking comment from me - just posing it for discussion.

Reply:
Well first when I read your comment I thought it makes sense to use HSL, I haven't thought about this at all. However I checked in Blender and Glaxnimate how they interpolate colors and it seems they are both not going through the HSL approach.

Reply:
HSL might be nice for hue, but saturation and lightness will still not wrap. Performance will be worse too, but it is unclear the impact. But since Blender and Glaxnimate do it this way, it might be preferable.

Reply:
I agree with this conclusion. Let's leave it as-is. Thanks for the discussion.